sw/host/opentitanlib/src/transport/verilator/subprocess.rs

```diff
@@ -0,0 +1,96 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::util::file;
+use anyhow::Result;
+use log::info;
+use regex::Regex;
+use std::io::Read;
+use std::process::{Child, ChildStdout, Command, Stdio};
+use std::time::{Duration, Instant};
+
+/// Verilator startup options.
+pub struct Options {
+    /// The verilator executable.
+    pub executable: String,
+    /// The ROM image used to boot the CPU.
+    pub rom_image: String,
+    /// The flash image stored in internal flash memory.
+    pub flash_image: String,
+    /// The OTP settings.
+    pub otp_image: String,
+    /// Any extra arguments to verilator.
+    pub extra_args: Vec<String>,
+}
+
+pub struct Subprocess {
+    child: Child,
+    stdout: ChildStdout,
+    accumulated_output: String,
+}
+
+impl Subprocess {
+    /// Starts a verilator [`Subprocess`] based on [`Options`].
+    pub fn from_options(options: Options) -> Result<Self> {
+        let mut command = Command::new(&options.executable);
+        let mut args = Vec::new();
+
+        if !options.rom_image.is_empty() {
+            args.push(format!("--meminit=rom,{}", options.rom_image));
+        }
+        if !options.flash_image.is_empty() {
+            args.push(format!("--meminit=flash,{}", options.flash_image));
+        }
+        if !options.otp_image.is_empty() {
+            args.push(format!("--meminit=otp,{}", options.otp_image));
+        }
+        args.extend_from_slice(&options.extra_args[..]);
```

Comment:
Nit: you don't need the [..] here, I think. Rust will deref-coerce `&Vec<T>` to `&[T]` in this situation (it only doesn't work if the function parameter is generic, which it isn't here).

Reply:
Done.